discovery/manager_test.go

```diff
@@ -868,40 +885,354 @@ scrape_configs:
 	}
 }
 
+func TestCoordinationWithReceiver(t *testing.T) {
+	updateDelay := 100 * time.Millisecond
+
+	type expect struct {
+		delay time.Duration
+		tgs   map[string][]*targetgroup.Group
+	}
+
+	testCases := []struct {
+		title     string
+		providers map[string]Discoverer
+		expected  []expect
+	}{
+		{
+			title: "Receiver should get an empty map even when the provider sends nothing and closes the channel",
```

Comment:
why do we need to test for this? Was there a bug for this?

Reply:
Not really. This is the result of some exploratory testing but agreed that it isn't necessary.